### PR TITLE
Feaature/sharing playlist

### PR DESCRIPTION
**Pull request by kkuepper**

### CODE REVIEW

File: BMM.UI.Android/Resources/layout/fragment_share_trackcollection.axml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
```
Comment (kkuepper) [inline BMM.UI.Android/Resources/layout/fragment_share_trackcollection.axml:2]: fragment_tracklist has pretty much the right design. Maybe it makes sense to reuse it?

File: BMM.UI.Android/Application/Fragments/ShareTrackCollectionFragment.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+using BMM.Core.ViewModels;
+using MvvmCross.Platforms.Android.Presenters.Attributes;
+
+namespace BMM.UI.Droid.Application.Fragments
+{
+    [MvxFragmentPresentation(typeof(MainActivityViewModel), Resource.Id.content_frame, true)]
+    [Register("bmm.ui.droid.application.fragments.ShareTrackCollectionFragment")]
+    public class ShareTrackCollectionFragment : BaseFragment<ShareTrackCollectionViewModel>
+    {
+        protected override int FragmentId => Resource.Layout.fragment_share_trackcollection;
```
Comment (kkuepper) [inline BMM.UI.Android/Application/Fragments/ShareTrackCollectionFragment.cs:14]: please have a look at tracklist

File: BMM.Core/ViewModels/Base/BaseViewModel.cs [hunk 3]
```diff
@@ -94,6 +97,22 @@ public IMvxCommand<Track> ShowTrackInfoCommand
             }
         }
 
+        private IMvxAsyncCommand _closeCommand;
```
[the remark this answers is not in the document]
Comment (akrol95) [reply BMM.Core/ViewModels/Base/BaseViewModel.cs:100]: as it could be used in any VM, it's worth to have such comand in BaseViewModel imo

**Review by kkuepper — COMMENTED** [review]
left some comments

File: BMM.UI.Android/Resources/layout/fragment_share_trackcollection.axml (hunk 1 above)
Comment (akrol95) [reply BMM.UI.Android/Resources/layout/fragment_share_trackcollection.axml:2]: I think you mean Shared Playlist view where we have list of songs/podcasts and for sure i want to reuse it, but is't not implemented yet. But in this case the fragment is showing Share Playlist view and I don't see any similiar view in the app, that's why I want to create separrate view for it.

File: BMM.UI.Android/Application/Fragments/ShareTrackCollectionFragment.cs (hunk 2 above)
Comment (akrol95) [reply BMM.UI.Android/Application/Fragments/ShareTrackCollectionFragment.cs:14]: Same as above

File: BMM.Core/ViewModels/Base/BaseViewModel.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (kkuepper) [reply BMM.Core/ViewModels/Base/BaseViewModel.cs:100]: Hm. We have 7 usages of _navigationService.Close().
In most cases we do additional stuff around the close operation (e.g. ask user if changes should be discarded). Only in PlayerViewModel we have exactly the same command.
I'm not sure if that is worth adding more code to the BaseViewModel for that.

[the remark this answers is not in the document]
Comment (akrol95) [reply BMM.Core/ViewModels/Base/BaseViewModel.cs:100]: I can remove it if you wish, but i think we should focus of cleaning base VM of some other less generic stuff than on command to close current VM

File: BMM.Core/ViewModels/Interfaces/IShareTrackCollectionViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+using BMM.Core.ViewModels.Parameters.Interface;
+using MvvmCross.ViewModels;
+
+namespace BMM.Core.ViewModels.Interfaces
+{
+    public interface IShareTrackCollectionViewModel : IMvxViewModel<ITrackCollectionParameter>
```
Comment (olivereisenhut) [inline BMM.Core/ViewModels/Interfaces/IShareTrackCollectionViewModel.cs:6]: Not sure about that interface. Could you shortly explain it to me?

File: BMM.UI.Android/Application/Activities/MainActivity.cs [hunk 5]
```diff
@@ -44,10 +46,11 @@ public class MainActivity : BaseFragmentActivity<MainActivityViewModel>
     {
         private IMediaPlayer _mediaPlayer;
         private AndroidMediaPlayer _androidPlayer;
+        private BottomNavigationView? _bottomNavigationView;
 
         private string _unhandledDeepLink;
 
-        protected override void OnCreate(Bundle bundle)
+        protected override async void OnCreate(Bundle bundle)
```
[the remark this answers is not in the document]
Comment (akrol95) [reply BMM.UI.Android/Application/Activities/MainActivity.cs:53]: some leftovers, i'll remove it

File: BMM.Core/ViewModels/Interfaces/IShareTrackCollectionViewModel.cs (hunk 4 above)
Comment (akrol95) [reply BMM.Core/ViewModels/Interfaces/IShareTrackCollectionViewModel.cs:6]: We should start adding interfaces to VMs if we want to use Guarded Actions in the future.

File: BMM.UI.iOS/Application/CustomViews/LeftImageButton.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
+using System;
+using System.ComponentModel;
+using CoreGraphics;
+using Foundation;
+using UIKit;
+
+namespace BMM.UI.iOS
+{
+    [Register(nameof(LeftImageButton)), DesignTimeVisible(true)]
```
[the remark this answers is not in the document]
Comment (akrol95) [reply BMM.UI.iOS/Application/CustomViews/LeftImageButton.cs:9]: If you mean if this button can appear in designer, then unfortunately not.

File: BMM.UI.iOS/Constants/Typography.cs [hunk 7]
```diff
@@ -20,12 +20,18 @@ public static class Typography
             return font;
         });
 
-        public static readonly Lazy<UIFont> Header2 = new Lazy<UIFont>(() =>
+        public static readonly Lazy<UIFont> Header1 = new Lazy<UIFont>(() =>
```
[the remark this answers is not in the document]
Comment (akrol95) [reply BMM.UI.iOS/Constants/Typography.cs:23]: Fixed

**Pull request closed without merging**